### ISSUE TITLE
[P2] Training-free architectural knobs: widen the sliding window, reallocate global layers

### ISSUE THREAD

**Issue by Parswanadh** [Parswanadh/laya-lab#4]
Owner: `engineer`. Charter `roles/README.md` §engineer. **Take the GPU flock.**

Read first: `findings/P1-mechanism.md` — the candidate mechanism. Your job is to test its two cheapest predictions. **No training.** Both are config-level architectural changes to the loaded encoder.

## Why this is the cheapest test of the whole thesis
P1 found the markers appear to have no privileged, non-diluting path to distant document positions: 14/22 layers are sliding-attention with half-window `local_attention // 2` = **64**, and only **8 global layers** (idx 0,3,6,9,12,15,18,21) create direct long-range edges.

## The exact mechanism (verified in the installed transformers 4.57.6)
`transformers/models/modernbert/modeling_modernbert.py`:
- **line 463-469** — at construction: `if layer_id % config.global_attn_every_n_layers != 0: self.local_attention = (config.local_attention // 2, config.local_attention // 2)` **else** `self.local_attention = (-1, -1)`.
- **line 412** — the SDPA path: `if local_attention != (-1, -1): attention_mask = sliding_window_mask`. `local_attention` is the **per-layer init-time** tuple.
- **line 938** — the mask is built per-forward from `self.config.local_attention`: `distance <= self.config.local_attention // 2`.

So the two knobs differ in how they must be applied **after** `laya.load()`:
- **Widening the window (P-a)** — the mask reads `config.local_attention` every forward, so setting `agent.model.encoder.config.local_attention = W` changes the mask immediately. Also set each sliding layer's `self_attn.local_attention = (W//2, W//2)` so the SDPA branch and the mask agree.
- **All-global (P-b)** — `self.local_attention` was fixed at init, so setting the config alone does **nothing**. You must also patch every layer: `lyr.self_attn.local_attention = (-1, -1)` and `lyr.attention_type = 'full_attention'`. **Verify the patch actually took effect** — assert on the mask, not on the attribute. A silent no-op here would produce a fake 'no effect' result.

## Arms to run
Use the frozen protocol in `protocol.md`. Report every arm against **majority-class (0.450)** and against the measured baselines (`findings/P1-mechanism.md` probe B/C).

1. **Baseline** — shipped config, limit=8192 (control; must reproduce probe B/C within noise).
2. **P-a** — `local_attention` ∈ {128 (control), 256, 512, 1024} at limit=8192.
3. **P-b** — all layers global (`global_attn_every_n_layers = 1` + layer patch) at limit=8192.
4. **Combination** — best of P-a with P-b, if P-b alone is not already best.

Report accuracy, median + p95 latency, and peak VRAM per arm. For P-b the latency will be much worse — **report the cost honestly**; a win that costs 10x is a different claim from a free win.

## Mandatory controls
- **Assert the intervention is live.** Before scoring, decode the max attended distance or compare attention masks between arms, and print it. If the mask did not change, the arm is invalid.
- **Ablation control.** Run `local_attention = 64` (a *narrower* window) too. If narrowing and widening both 'help', something other than the window is moving the number.
- **Short-context regression.** Score pad=0 in every arm. Widening or globalising attention **must not** damage the short-context case. A gain at 7000 that costs the pad=0 case is not shippable.

## Outputs
- `experiments/h2h3-knobs/manifest.json` + raw per-item JSONL.
- `findings/E-002.md` — the table, with `n`, the assertion that the patch was live, and an explicit verdict on P-a and P-b (**refuted is a valid and valuable verdict**).
- Push as you go. Do **not** modify `fork/laya/`. This is measurement only.
